pkg/storageos/configmap.go

```diff
@@ -121,19 +126,89 @@ const (
 	tlsEtcdCertsVolume = "etcd-certs"
 )
 
-// createService creates a ConfigMap to store the node container configuration.
-func (s *Deployment) createConfigMap() error {
+// ensureConfigMap creates or updates a ConfigMap to store the node container
+// configuration.
+func (s *Deployment) ensureConfigMap() error {
 	config := configFromSpec(s.stos.Spec, CSIV1Supported(s.k8sVersion))
 
 	labels := make(map[string]string)
```

Comment:
What's going on here? 

It looks like the call to `s.k8sResourceManager.ConfigMap(configmapName, s.stos.Spec.GetResourceNS(), labels, config).Get()` calls `ResourceManager.combineLabels()` which creates a new map and combines the `ResourceManager.Labels` with the labels that are passed in. 

Would passing in nil risk causing an explosion later because we'd instantiate r with a nil map for it's labels?

Reply:
Should be just fine to pass nil. The checks are handled internally.

Reply:
No, combineLabels checks for nil and returns r.labels, which is never nil since it's set in the constructor.  So passing in nil is ok, and it would be safe to remove `labels := make(map[string]string)`. I'd prefer to keep it as I think it makes it more clear that we don't want to set specific labels.